src/script/lua_api/l_object.cpp

```diff
@@ -863,12 +863,21 @@ int ObjectRef::l_add_velocity(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	ObjectRef *ref = checkobject(L, 1);
-	LuaEntitySAO *co = getluaobject(ref);
-	if (!co)
+	v3f vel = checkFloatPos(L, 2);
+
+	ServerActiveObject *obj = getobject(ref);
+	if (obj == nullptr)
 		return 0;
-	v3f pos = checkFloatPos(L, 2);
-	// Do it
-	co->addVelocity(pos);
+
+	if (obj->getType() == ACTIVEOBJECT_TYPE_LUAENTITY) {
+		LuaEntitySAO *co = dynamic_cast<LuaEntitySAO*>(obj);
+		co->addVelocity(vel);
+	} else if (obj->getType() == ACTIVEOBJECT_TYPE_PLAYER) {
+		PlayerSAO *player = dynamic_cast<PlayerSAO*>(obj);
+		player->setMaxSpeedOverride(vel);
+		getServer(L)->SendPlayerSpeed(player->getPeerID(), vel);
```

Comment:
This is the same behaviour as before. The function names feel wrong though, it should be `SendPlayerAddSpeed`. Surely maxSpeedOverride should be relative too?

Reply:
relative to what?

Reply:
To the current velocity, although that's not accurately possible due to network latency

Reply:
`SendPlayerAddSpeed` or `SendIncreasePlayerSpeed` would both be fine for me. Renaming this function is not necessary IMO, but feel free to do so.